pennylane_lightning/src/tests/Test_Cache.cpp

```diff
@@ -0,0 +1,289 @@
+#include <catch2/catch.hpp>
+
+#include "Cache.hpp"
+
+using namespace Pennylane::Util;
+
+// Boost implementation of a hash combine:
+// https://www.boost.org/doc/libs/1_35_0/doc/html/boost/hash_combine_id241013.html
+struct hash_function {
+    std::size_t
+    operator()(const std::pair<std::vector<size_t>, size_t> &key) const {
+        std::size_t combined_hash_value = 0;
+        static const size_t offset = 0x9e3779b9;
+
+        for (const auto &term : key.first) {
+            combined_hash_value ^= std::hash<size_t>()(term) + offset +
+                                   (combined_hash_value << 6) +
+                                   (combined_hash_value >> 2);
+        };
+        combined_hash_value ^= std::hash<size_t>()(key.second) + offset +
+                               (combined_hash_value << 6) +
+                               (combined_hash_value >> 2);
+        return combined_hash_value;
+    }
+};
+
+// NOLINTNEXTLINE
+TEST_CASE("LRU_Cache for doubles, with size_t keys", "[Cache]") {
+    LRU_cache<size_t, double> cache_container{10};

```

Comment:
Can we add a test also for a 0-sized container, to test that the cache does not operate? In this case, it should default to the current behaviour with no caching.

Reply:
I commit your suggestion to default the cache size to 0. I implemented the cache_size=0 to be the no-cache scenario when you suggested it sometime during the beginning of this PR.  I will implement tests for the 0-cache case as soon as I can.